clinica/iotools/converters/oasis3_to_bids/oasis3_to_bids.py

```diff
@@ -0,0 +1,165 @@
+                else:
+                    sessions_dict[y][z]["diagnosis"] = "CN"
+
+        bids.write_sessions_tsv(bids_dir, sessions_dict)
+
+        # --Create scans files--
+        # Note: We have no scans information for OASIS
+        scans_dict = bids.create_scans_dict(
+            clinical_data_dir,
+            "OASIS",
+            clinic_specs_path,
+            bids_ids,
+            "ID",
+            "",
+            sessions_dict
+        )
+        bids.write_scans_tsv(bids_dir, bids_ids, scans_dict)
+
+    def convert_images(self, source_dir, dest_dir):
+        """Convert T1w images to BIDS.
+
+        Args:
+            source_dir: path to the OASIS dataset
+            dest_dir: path to the BIDS directory
+        """
+        from os import path
+        from glob import glob
+        import os
+
+        def convert_single_subject(subj_folder):
+            import os
+            import re
+            import shutil
+
+            def copy_file(source, target):
+                try:
+                    new_filename = re.sub("OAS3", "OASIS3", path.basename(source))
+                    new_filename = re.sub("_echo-", "_run-", new_filename)
+                    if re.search(".*_pet.(nii.gz|json)", new_filename) is not None:
+                        new_filename = re.sub("task-rest_", "", new_filename)
+                        new_filename = re.sub("acq", "trc", new_filename)
+                        target = path.join(path.dirname(target), "pet")
+                    elif re.search(".*_(minIP|swi|dwi).(nii.gz|json)", new_filename) is not None:
+                        target = path.join(path.dirname(target), "swi")
+                    if not os.path.isdir(target):
+                        os.mkdir(path.join(target))
+                    shutil.copy(source, target)
+                    os.rename(
+                        path.join(target, path.basename(source)),
+                        path.join(target, new_filename)
+                    )
+                except IOError as e:
+                    print("Unable to copy file. %s" % e)
+
+            subj_id = os.path.basename(subj_folder)
+            print("Converting ", subj_id)
+            old_sess_fold = subj_id.split("_")
+            numerical_id = re.sub(r'^OAS3', '', old_sess_fold[0])
+            bids_id = "sub-OASIS3" + str(numerical_id)
+            bids_subj_folder = path.join(dest_dir, bids_id)
+            if not os.path.isdir(bids_subj_folder):
+                os.mkdir(bids_subj_folder)
+
+            session_name = old_sess_fold[2]
+            session_folder = path.join(bids_subj_folder, "ses-" + session_name)
+            if not os.path.isdir(session_folder):
+                os.mkdir(path.join(session_folder))
+
+            # In order do convert the Analyze format to Nifti the path to the .img file is required
+            nifti_folders = glob(
+                path.join(subj_folder, "**/*.nii.gz"), recursive=True
+            )
```

Comment:
I don't understand. Your comment says you are looking for folders containing an `.img`, yet you're doing a glob search on files ending with `.nii.gz`. This is confusing. Besides `nifti_folders` is not a good name since it will contain file paths, not folders.

Reply:
Older comments for OASIS which the OASIS3 code was largely inspired by. I didn't dare touch these comments too much without a reviewing. Indeed, it's on nifti paths. I could have done it for this one since it was obsolete.

Reply:
Then let's remove the comment and rename the variable to `nifti_files`